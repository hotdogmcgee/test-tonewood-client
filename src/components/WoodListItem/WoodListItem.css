.WoodListItem {
  background: none;
  display: flex;
  flex-direction: column;
  border: 0;
  font-size: smaller;
  text-decoration: none;
  box-shadow: 0px 11px 21px -8px rgba(0, 0, 0, 0.5);
  border: 4px solid grey;
  margin-bottom: 40px;
  border-radius: 10px;
  text-decoration: none !important;
}

.WoodListItem__heading {
  font-weight: 500;
}

.WoodListItem:hover {
  cursor: pointer;
}

.WoodListItem,
.WoodListItem:visited {
  color: black;
}

.WoodListItem__details {
  padding: 10px;
  display: flex;
  flex: 1;
  flex-direction: column;
  justify-content: space-between;
  width: 275px;
}

.WoodListItem__details h2 {
  margin: 0;
  font-size: 24px;
}

.WoodListItem__details span {
  display: block;
  text-align: center;
  opacity: 0.5;
}

.scientific-name {
  font-style: italic;
}

@media screen and (min-width: 900px) {
  .WoodListItem {
    min-height: 150px;
  }
  .WoodListItem__details h2 {
    font-size: 32px;
  }
}
