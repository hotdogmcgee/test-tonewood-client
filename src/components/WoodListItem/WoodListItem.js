import React from "react";
import { Link } from "react-router-dom";
import "./WoodListItem.css";

export default class WoodListItem extends React.Component {
  render() {
    const { wood } = this.props;
    return (
      <Link to={`/woods/${wood.id}`} className="WoodListItem">
        {/* <div className='WoodListItem__image' style={{backgroundImage: '../../WoodImages/ThujaPlicata.jpg'}} /> */}

        <div className="WoodListItem__details">
          <div className="WoodListItem__text">
            <h2 className="WoodListItem__heading">{wood.common_name}</h2>
            <p className="scientific-name">
              {wood.genus} {wood.species}
            </p>
            {/* <p className='WoodListItem__description'>{truncate(wood.description)}</p> */}
          </div>

          <div className="WoodListItem__submissions">
            <span id="WoodListItem_submission-count">
              {readableSubmissionCount(wood.number_of_submissions)}
            </span>
          </div>
        </div>
      </Link>
    );
  }
}

function readableSubmissionCount(number) {
  switch (number) {
    case 0:
      return "no submissions yet";

    case 1:
      return "1 submission";

    default:
      return `${number} submissions`;
  }
}

