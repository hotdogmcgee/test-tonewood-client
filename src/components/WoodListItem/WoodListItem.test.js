import React from "react";
import WoodListItem from "./WoodListItem";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("WoodListItem component", () => {

    const wood = {
        id: 2,
        common_name: 'maple',
        genus: 'maple',
        species: 'wood',
        number_of_submissions: 4
    }
  it("renders a WoodListItem by default", () => {
    const wrapper = shallow(<WoodListItem />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders a WoodListItem given props', () => {
    const wrapper = shallow(<WoodListItem wood={wood}/>);
    expect(toJson(wrapper)).toMatchSnapshot()
  })
});