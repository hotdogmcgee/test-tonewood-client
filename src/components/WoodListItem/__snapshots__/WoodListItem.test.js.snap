// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WoodListItem component renders a WoodListItem by default 1`] = `""`;

exports[`WoodListItem component renders a WoodListItem given props 1`] = `
<Link
  className="WoodListItem"
  to="/woods/2"
>
  <div
    className="WoodListItem__details"
  >
    <div
      className="WoodListItem__text"
    >
      <h2
        className="WoodListItem__heading"
      >
        maple
      </h2>
      <p
        className="scientific-name"
      >
        maple
         
        wood
      </p>
    </div>
    <div
      className="WoodListItem__submissions"
    >
      <span
        id="WoodListItem_submission-count"
      >
        4 submissions
      </span>
    </div>
  </div>
</Link>
`;
